Shared/Samples/Find route around barriers/FindRouteAroundBarriersView.Model.swift

```diff
@@ -0,0 +1,212 @@
+        // MARK: Methods
+        
+        /// Adds a stop graphic to the map.
+        /// - Parameter point: The point to add the stop graphic at.
+        func addStopGraphic(at point: Point) {
+            // Create a text symbol with the next index.
+            let textSymbol = TextSymbol(
+                text: "\(stopsCount + 1)",
+                color: .white,
+                size: 20,
+                horizontalAlignment: .center,
+                verticalAlignment: .middle
+            )
+            textSymbol.offsetY = stopSymbol.offsetY
+            
+            // Create a graphic with the marker symbol and text symbol.
+            let compositeSymbol = CompositeSymbol(symbols: [stopSymbol, textSymbol])
+            let stopGraphic = Graphic(geometry: point, symbol: compositeSymbol)
+            
+            // Add the new graphic to the stop graphics overlay.
+            stopGraphicsOverlay.addGraphic(stopGraphic)
+            stopsCount += 1
+        }
+        
+        /// Adds a barrier graphic to the map.
+        /// - Parameter point: The point to add the barrier graphic at.
+        func addBarrierGraphic(at point: Point) {
+            // Buffer the point and create the barrier symbol.
+            let bufferedGeometry = GeometryEngine.buffer(around: point, distance: 500)
+            let barrierSymbol = SimpleFillSymbol(style: .diagonalCross, color: .red)
+            
+            // Create a graphic from the symbol and buffer and add it to the graphics overlay.
+            let barrierGraphic = Graphic(geometry: bufferedGeometry, symbol: barrierSymbol)
+            barrierGraphicsOverlay.addGraphic(barrierGraphic)
+            barriersCount += 1
+        }
+        
+        /// Resets all the features on the map associated with a given feature type.
+        /// - Parameter features: The features to remove from the map.
+        func reset(features: RouteFeatures) {
+            if features == .stops {
+                // Reset the stops.
+                stopGraphicsOverlay.removeAllGraphics()
+                stopsCount = 0
+                
+                // Reset the route.
+                route = nil
+                routeGraphic.geometry = nil
+                directionGraphic.geometry = nil
+                updateRouteInfoText()
+            } else {
+                barrierGraphicsOverlay.removeAllGraphics()
+                barriersCount = 0
+            }
+        }
+        
+        /// Routes using the route parameters and the current stops and barriers on the map.
+        func route() async throws {
+            // Update the route parameters' stops using the stop graphics.
+            var stops = [Stop]()
+            for stopGraphic in stopGraphicsOverlay.graphics {
+                guard let point = stopGraphic.geometry as? Point else { continue }
+                let stop = Stop(point: point)
+                stops.append(stop)
+            }
+            routeParameters.setStops(stops)
```

Comment:
I've been advised to try to avoid imperative for loops in Swift and to prefer, where possible, to use functional style programming constructs:

```suggestion
            routeParameters.setStops(
                stopGraphicsOverlay.graphics
                    .compactMap { $0.geometry as? Point }
                    .map(Stop.init(point:))
            )
```

Same feedback applies elsewhere.